eldritchmush/commands/combatant.py

```diff
@@ -274,3 +278,98 @@ def hasChirurgeonsKit(self):
     def useChirurgeonsKit(self):
         self.helper.useKit()
 
+    def takeShieldDamage(self, amount):
+        return self.alternateDamage(amount, 'shield_value')
+
+    def takeArmorSpecialistDamage(self, amount):
+        return self.alternateDamage(amount, "armor_specialist")
+
+    def takeArmorDamage(self, amount):
+        return self.alternateDamage(amount, "tough")
+
+    def takeToughDamage(self, amount):
+        return self.alternateDamage(amount, "armor")
+
+    def alternateArmorSpecialistDamage(self, amount):
+        return self.alternateDamage(amount, "armor_specialist")
+
+    def takeBodyDamage(self, amount):
+        return self.alternateDamage(amount, "body")
+
+    def takeBleedDamage(self, amount):
+        return self.alternateDamage(amount, "bleed_points")
+
+    def takeDeathDamage(self, amount, combatant):
+        if self.caller.db.creature_color:
+            weapon = combatant.getWeapon()
+            if weapon and (weapon.db.color == self.caller.db.creature_color):
+                return self.alternateDamage(amount, "death_points")
+            else:
+                return 0
+        else:
+            return self.alternateDamage(amount, "death_points")
+
+    def alternateDamage(self, amount, type):
+        remaining_damage = amount
+
+        if self.caller.attributes.get(type):
+            if self.caller.attributes.get(type):
+                # How much damage is left after the shield
+                remaining_damage = amount - self.caller.attributes.get(type)
+                if remaining_damage < 0:
+                    remaining_damage = 0
+
+                # Damage the shield
+                if amount >= self.caller.attributes.get(type):
+                    new_value = 0
+                else:
+                    new_value = self.caller.attributes.get(type) - amount
+
+                obj_to_set = self.caller.attributes.get(type, return_obj=True)
+                obj_to_set.value = new_value
+
+        return remaining_damage
+
+    def updateAv(self):
+        self.caller.db.av = self.caller.db.shield_value + self.caller.db.armor + self.caller.db.tough + self.caller.db.armor_specialist
```

Comment:
You need to add indomitable to this calc and anywhere you update av

Reply:
Cool, I was just grabbing from Code so it’s probably missing in a few places, should indomitable get decreased like tough when you take damage?